From d8759969fe73cb6bcbeb53bae0d9bd8f88eb64bd Mon Sep 17 00:00:00 2001
From: bagyusz <phelyx92@gamil.com>
Date: Sat, 22 Aug 2015 19:32:30 +0200
Subject: [PATCH] [REVERT]:"display: Add support for interleaved YUY2 and YUYV 
 format. Add support for interleaved HAL_PIXEL_FORMAT_YCbCr_422_I and 
 HAL_PIXEL_FORMAT_YCrCb_422_I format in display HAL.

Change-Id: I820e415b6735a62ebe03d909b98d1149ed9805ad
---
 libcopybit/copybit.cpp          | 2 --
 libgralloc/alloc_controller.cpp | 4 ----
 libgralloc/gralloc_priv.h       | 1 -
 libhwcomposer/Android.mk        | 2 +-
 liboverlay/overlayUtils.cpp     | 8 --------
 liboverlay/overlayUtils.h       | 3 ---
 6 files changed, 1 insertion(+), 19 deletions(-)

diff --git a/libcopybit/copybit.cpp b/libcopybit/copybit.cpp
index 5f48e07..39d963d 100644
--- a/libcopybit/copybit.cpp
+++ b/libcopybit/copybit.cpp
@@ -127,8 +127,6 @@ static int get_format(int format) {
         case HAL_PIXEL_FORMAT_RGB_888:       return MDP_RGB_888;
         case HAL_PIXEL_FORMAT_RGBA_8888:     return MDP_RGBA_8888;
         case HAL_PIXEL_FORMAT_BGRA_8888:     return MDP_BGRA_8888;
-        case HAL_PIXEL_FORMAT_YCrCb_422_I:   return MDP_YCRYCB_H2V1;
-        case HAL_PIXEL_FORMAT_YCbCr_422_I:   return MDP_YCBYCR_H2V1;
         case HAL_PIXEL_FORMAT_YCrCb_422_SP:  return MDP_Y_CRCB_H2V1;
         case HAL_PIXEL_FORMAT_YCrCb_420_SP:  return MDP_Y_CRCB_H2V2;
         case HAL_PIXEL_FORMAT_YCbCr_422_SP:  return MDP_Y_CBCR_H2V1;
diff --git a/libgralloc/alloc_controller.cpp b/libgralloc/alloc_controller.cpp
index 46f6f9b..d995dfb 100644
--- a/libgralloc/alloc_controller.cpp
+++ b/libgralloc/alloc_controller.cpp
@@ -146,8 +146,6 @@ int AdrenoMemInfo::getStride(int width, int format)
             case HAL_PIXEL_FORMAT_YV12:
             case HAL_PIXEL_FORMAT_YCbCr_422_SP:
             case HAL_PIXEL_FORMAT_YCrCb_422_SP:
-            case HAL_PIXEL_FORMAT_YCbCr_422_I:
-            case HAL_PIXEL_FORMAT_YCrCb_422_I:
                 stride = ALIGN(width, 16);
                 break;
             case HAL_PIXEL_FORMAT_YCbCr_420_SP_VENUS:
@@ -357,8 +355,6 @@ size_t getBufferSizeAndDimensions(int width, int height, int format,
             break;
         case HAL_PIXEL_FORMAT_YCbCr_422_SP:
         case HAL_PIXEL_FORMAT_YCrCb_422_SP:
-        case HAL_PIXEL_FORMAT_YCbCr_422_I:
-        case HAL_PIXEL_FORMAT_YCrCb_422_I:
             if(width & 1) {
                 ALOGE("width is odd for the YUV422_SP format");
                 return -EINVAL;
diff --git a/libgralloc/gralloc_priv.h b/libgralloc/gralloc_priv.h
index 58fc898..7d3f983 100644
--- a/libgralloc/gralloc_priv.h
+++ b/libgralloc/gralloc_priv.h
@@ -104,7 +104,6 @@ enum {
     HAL_PIXEL_FORMAT_RG_88                  = 0x10E,
     HAL_PIXEL_FORMAT_YCbCr_444_SP           = 0x10F,
     HAL_PIXEL_FORMAT_YCrCb_444_SP           = 0x110,
-    HAL_PIXEL_FORMAT_YCrCb_422_I            = 0x111,
     HAL_PIXEL_FORMAT_NV21_ZSL               = 0x112,
     HAL_PIXEL_FORMAT_INTERLACE              = 0x180,
 
diff --git a/libhwcomposer/Android.mk b/libhwcomposer/Android.mk
index 8bd5608..b197389 100644
--- a/libhwcomposer/Android.mk
+++ b/libhwcomposer/Android.mk
@@ -12,7 +12,7 @@ LOCAL_SHARED_LIBRARIES        := $(common_libs) libEGL liboverlay \
                                  libexternal libqdutils libhardware_legacy \
                                  libdl libmemalloc libqservice libsync \
                                  libbinder libmedia libvirtual libskia
-LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"qdhwcomposer\"
+LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"qdhwcomposer\" -Os
 LOCAL_ADDITIONAL_DEPENDENCIES := $(common_deps)
 LOCAL_SRC_FILES               := hwc.cpp          \
                                  hwc_utils.cpp    \
diff --git a/liboverlay/overlayUtils.cpp b/liboverlay/overlayUtils.cpp
index 64a0bb6..af68cb9 100644
--- a/liboverlay/overlayUtils.cpp
+++ b/liboverlay/overlayUtils.cpp
@@ -114,10 +114,6 @@ int getMdpFormat(int format) {
             return MDP_Y_CBCR_H2V2;
         case HAL_PIXEL_FORMAT_YCrCb_422_SP:
             return MDP_Y_CRCB_H2V1;
-        case HAL_PIXEL_FORMAT_YCbCr_422_I:
-            return MDP_YCBYCR_H2V1;
-        case HAL_PIXEL_FORMAT_YCrCb_422_I:
-            return MDP_YCRYCB_H2V1;
         case HAL_PIXEL_FORMAT_YCbCr_444_SP:
             return MDP_Y_CBCR_H1V1;
         case HAL_PIXEL_FORMAT_YCrCb_444_SP:
@@ -170,10 +166,6 @@ int getHALFormat(int mdpFormat) {
             return HAL_PIXEL_FORMAT_YCbCr_420_SP;
         case MDP_Y_CRCB_H2V1:
             return HAL_PIXEL_FORMAT_YCrCb_422_SP;
-        case MDP_YCBYCR_H2V1:
-            return HAL_PIXEL_FORMAT_YCbCr_422_I;
-        case MDP_YCRYCB_H2V1:
-            return HAL_PIXEL_FORMAT_YCrCb_422_I;
          case MDP_Y_CBCR_H1V1:
             return HAL_PIXEL_FORMAT_YCbCr_444_SP;
         case MDP_Y_CRCB_H1V1:
diff --git a/liboverlay/overlayUtils.h b/liboverlay/overlayUtils.h
index 231887c..ad4dd87 100644
--- a/liboverlay/overlayUtils.h
+++ b/liboverlay/overlayUtils.h
@@ -483,8 +483,6 @@ inline bool isYuv(uint32_t format) {
         case MDP_Y_CR_CB_H2V2:
         case MDP_Y_CR_CB_GH2V2:
         case MDP_Y_CBCR_H2V2_VENUS:
-        case MDP_YCBYCR_H2V1:
-        case MDP_YCRYCB_H2V1:
             return true;
         default:
             return false;
@@ -514,7 +512,6 @@ inline const char* getFormatString(int format){
         "MDP_ARGB_8888",
         "MDP_RGB_888",
         "MDP_Y_CRCB_H2V2",
-        "MDP_YCBYCR_H2V1",
         "MDP_YCRYCB_H2V1",
         "MDP_Y_CRCB_H2V1",
         "MDP_Y_CBCR_H2V1",
-- 
1.9.1

